rfcs/syntax-property-access-modifiers.md

```diff
@@ -0,0 +1,113 @@
+# Feature name
+
+## Summary
+
+We need syntax to match the semantics of read-only and writed-only modifiers for table properties and indexers.
+
+## Motivation
+
+See the semantic RFCs for motivation.
+
+## Design
+
+TBD
+
+## Drawbacks
+
+TBD
+
+## Alternatives
+
+The design space for syntax includes:
+
+* Names, symbols or attributes?
+* Modifier position?
+* How to parse or serialize properties with differing read- and write-types?
+
+### Names, symbols or attributes?
+
+We could use names for modifiers, such as
+
+* `get` and `set`
+* `const` and `mut`
+* `read` and `write`
+* `readonly` and `writeonly`
+
+One issue is that these are all valid identifiers, so if we want
+backward compatbility, they cannot be made keywords. This presents
+issues with code that uses the chosen names as type or property names,
+for instance:
+
+```lua
+  type set = { [any] : bool }
+  type ugh = { get set : set }
+```
+
+We could use symbols, for example
+
+* `+` and `-`
+* (Are there other obvious pairs of symbols?)
+
+We could use attributes, for example
+
+* `@get` and `@set`
+* ... as per "names for modifiers", only prefixed by `@` ...
+
+These both have the advantage of being unambiguous and easier to
+parse. Symbols are terser, whch is both good and bad.
```

Comment:
Adding a new sigil to Luau is probably not the best course of action here. I vote against `@`-attributes.

Reply:
We may end up with attribute for things like API documentation, deprecation, etc. The syntax which most languages seem to have adopted is `@attribute`.

Reply:
(Sorry for the delay, it's been the winter break!)